### ISSUE TITLE
Failed to parse logger rotate text

### ISSUE THREAD

**Issue by Good-Pudge** [pocoproject/poco#2424]
#### Expected behavior
`# Log file created/rotated Sunday, 12 Aug 18 13:35:39 GMT` string is should parsed successfully.

#### Actual behavior
But I get exception with message:
```
Syntax error: Invalid DateTimeString:Sunday, 12 Aug 18 13:35:39 GMT
```

#### Steps to reproduce the problem
Compile and start code:
```cpp
#include <Poco/Logger.h>
#include <Poco/FileChannel.h>
#include <Poco/PatternFormatter.h>
#include <Poco/FormattingChannel.h>


using namespace Poco;


int main()
{
	Channel::Ptr file = new FileChannel("log/app.log");
	file->setProperty("rotation", "daily");

	Formatter::Ptr formatter = new PatternFormatter("[%p] %H:%M:%S.%i %d.%m.%Y %Z: %t");

	FormattingChannel::Ptr channel = new FormattingChannel;
	channel->setChannel(file);
	channel->setFormatter(formatter);

	auto& logger = Logger::root();
	logger.setChannel(channel);

	logger.fatal("Test");
}

```

#### POCO version
`develop` branch

#### Compiler and version
MinGW-w64 x64 8.1.0, posix threads, seh exception handling
VS 2017 MSVC 19.14.26430.0

#### Operating system and version
Windows 10 Pro x64 1803

#### Other relevant information
`log/app.log` file:
```
# Log file created/rotated Sunday, 12 Aug 18 13:35:39 GMT
[Fatal] 13:35:39.322 12.08.2018 GMT: Test
```

**Comment by Good-Pudge** [comment]
I don't think that it is MinGW problem. I wrote in main comment that I tested in MSVC too.

**Comment by aleks-f** [comment]
ok, can you step through it in debugger and see where the problem exactly is?

**Comment by Good-Pudge** [comment]
Ok, I'll do it.

**Comment by Good-Pudge** [comment]
I tested it on Linux and I can't reproduce problem. Now I'll use debugger in MinGW.

**Comment by Good-Pudge** [comment]
I found error. `std::string DateTimeFormat::RFC1036_REGEX` is invalid:
```
"(((Monday)|(Tuesday)|(Wednesday)|(Thursday)|(Friday)|(Saturday)|(Sun)), *)?"
"\\d\\d? +"
"((Jan)|(Feb)|(Mar)|(Apr)|(May)|(Jun)|(Jul)|(Aug)|(Sep)|(Oct)|(Nov)|(Dec)) +"
"\\d\\d(\\d\\d)? +\\d\\d:\\d\\d(:\\d\\d)? "
"((UT)|(GMT)|(EST)|(EDT)|(CST)|(CDT)|(MST)|(MDT)|(PST)|(PDT)|)?+"
"(([+\\-]?\\d\\d\\d\\d)?|(UT)|(GMT)|(EST)|(EDT)|(CST)|(CDT)|(MST)|(MDT)|(PST)|(PDT)|\\w)
```
As you can see there is written `Sun` instead of `Sunday`.
Also invalid argument is passed to `DateTimeParser::parse`. It look like `Sunday, 12 Aug 18 13:35:39 GMT\r`. Line break is `\r\n` in Windows, `\n` in UNIX. Because of it we can't reproduce problem in macOS, Linux, etc.

**Comment by aleks-f** [comment]
regex fixed, can you fix the line ending?

**Comment by aleks-f** [comment]
as for `Wdy`, the way I understand it, it means "weekday", not "Wednesday"

**Comment by Good-Pudge** [comment]
Hm... As I can see, RFC1036 look like `Mon, 15 Aug 05 15:52:01 +0000` in [PHP](https://secure.php.net/manual/en/class.datetime.php) and like `EEEEEEEEE, dd-MMM-yy HH:mm:ss z` in [Java](http://www.java2s.com/Tutorial/Java/0040__Data-Type/RFCdateformat.htm). Ok, doesn't matter.

About fix. We need to introduce our own function as cross-platform analog into `File` or `FileStream` probably.

**Comment by Good-Pudge** [comment]
`std::getline` accept just `char` delimiter type. Oh, `LineEndingConverter` exactly what is needed I think.
